### DISCUSSION TITLE
[Self-answered] Associating a Tiled Collider object Actor back to its object?

### DISCUSSION THREAD

**Discussion by DaVince** [excaliburjs/Excalibur#2508] [Q&A]
**I basically answered my own question by rubberducking in this very thread so I no longer need an answer (unless there's a better way to do this!) but I'm still going to post it in case any future person with the same question comes across this.**

![image](https://user-images.githubusercontent.com/105533/197405307-8760f347-80c3-4664-bb51-3781f61b396d.png)

I have set up various of my Tiled objects to be passive BoxColliders recognized by Excalibur, with the aim to have them as triggers to various actions that the game should then perform.

That said, when I collide with them, I get back an Actor object that Excalibur creates automatically. The only way to relate this back to the matching `ExcaliburObject` containing the custom properties that I'd like to access, seems to be through the object's name.

I was hoping to use the object name to define the object's behavior (e.g. several objects with the same name like "text" or "warp") but it looks like I'm going to have to give them unique identifiers instead.

**Comment by kamranayub** [comment]
@eonarheim is this something that is addressed in the tiled parser rewrite? It would seem helpful to either:

1. pass a reference to the underlying `ExcaliburObject` from `TiledResource` (I assume that's where it got made)
2. make it easy to look up via unique ID (I see Tiled gives us an ID, maybe a map from that to Excalibur ID?)

QOL-wise, 1. seems nicer.

`this.getComponent(Tiled).*` or something...
